src/main/kotlin/net/yested/core/properties/properties.kt

```diff
@@ -13,24 +14,39 @@ private interface PropertyChangeListener<in T> {
 interface ReadOnlyProperty<out T> {
     fun get():T
     fun onNext(handler: (T)->Unit):Disposable
+    fun detectContentChange()
 }
 
 class Property<T>(initialValue: T): ReadOnlyProperty<T> {
 
     private var value : T = initialValue
-    private var valueHashCode = value?.hashCode()
+    private var valueHash = getValueHash(value)
     private val listeners = mutableSetOf<PropertyChangeListener<T>>()
 
     fun set(newValue: T) {
-        // check the hashCode in case the value is mutable, the same instance is provided, but its contents have changed.
-        val newValueHashCode = newValue?.hashCode()
-        if (newValue != value || newValueHashCode != valueHashCode) {
+        // check the value hash in case the value is mutable, the same instance is provided, but its contents have changed.
+        val newValueHash = getValueHash(newValue)
+        if (newValue != value || newValueHash != valueHash) {
```

Comment:
I don't put new instances every time.  This is to handle mutable lists where the instance doesn't change.  By storing the hash, when the list changes, and its hash changes, the change can be detected.